src/libecalc/core/consumers/compressor/component.py

```diff
@@ -72,12 +72,20 @@ def evaluate(
         """
         self._operational_settings = operational_settings
 
+        # Creating a single input rate until we decide how to deal with multiple rates, multiple rates added because of
+        # multiple streams and pressures model, but we need to look at how streams are defined there.
+        total_requested_rate = TimeSeriesRate(
+            timesteps=operational_settings.timesteps,
+            values=list(np.sum([rate.values for rate in operational_settings.stream_day_rates], axis=0)),
+            unit=operational_settings.stream_day_rates[0].unit,
+            regularity=operational_settings.stream_day_rates[0].regularity,
+            rate_type=operational_settings.stream_day_rates[0].rate_type,
+        )
+
         model_results = []
         evaluated_timesteps = []
 
-        # TODO: This is a false assumption and will be dealt with shortly (that the regularity is the same
-        #   for all timesteps, and only taken for the first timestep). Not the first timestep, the first rate
-        evaluated_regularity = operational_settings.stream_day_rates[0].regularity
+        evaluated_regularity = total_requested_rate.regularity
```

Comment:
so the one above to restrict to one stream is to 1. handle regularity correctly and 2. fix multiple streams problem later? (to be discussed?)

Reply:
It's the same we have been doing. I moved the total_requested_rate to the top to make it a bit more clear, and reusable. So we don't have several places where we sum the rates. Regularity is kind of weird, because we are relying on it being set on installation, i.e. the same for all vectors. But that is a TODO. 

And yes, I have postponed dealing with multiple streams here, which will not work correctly with a single rate. To be discussed